src/geminim.nim

```diff
@@ -139,8 +139,17 @@ proc processTitanRequest(server: Server, req: Request): Future[Response] {.async
     return response(StatusNotAuthorised, "Token not recognized")
 
   var filePath = req.res.filePath
-  if dirExists(filePath):
-    filePath = filePath / "index.gmi" # assume we want to write index.gmi
+  let (parent, _ )= filePath.splitPath
+
+  if dirExists(parent):
+    if dirExists(filePath): # We're writing index.gmi in an existing directory
+      filePath = filePath / "index.gmi"
+  else: # we're writing a file in a new directory
+    try:
+      createDir(parent)
+    except:
```

Comment:
This should only catch `OSError` for safety.
```nim
except OSError
```

Reply:
It turns out that in the event of being unable to create the folder because of it actually being a file, it actually throws an `IOError`, rather than an `OSError` as documented. I've added a check for that as well.